TESTING.asciidoc

```diff
@@ -91,23 +91,22 @@ Run any test methods that contain 'esi' (like: ...r*esi*ze...).
 
 You can also filter tests by certain annotations ie:
 
-  * `@Nightly` - tests that only run in nightly builds (disabled by default)
   * `@Backwards` - backwards compatibility tests (disabled by default)
```

Comment:
We don't actually use `@Backwards` or `@BadApple` either! At this point we really only use `@AwaitsFix` I think.

Reply:
I left the `@BadApple` and `@Backwards` so that the filtering examples with more complicated boolean operators and grouping are valid. But you are right, as only `@AwaitsFix` is still used, the other annotations should be removed as well. 

I was thinking to remove the `@Backwards` separately as there is also a section using the backwards filtering which I guess is outdated: https://github.com/elastic/elasticsearch/blob/ad26075e989de81df61ec9f2eabce2bbc00a4896/TESTING.asciidoc#backwards-compatibility-tests
and should be replaced by:
https://github.com/elastic/elasticsearch/blob/ad26075e989de81df61ec9f2eabce2bbc00a4896/TESTING.asciidoc#testing-backwards-compatibility
I simply did not want to mix it all together ;) 

Just let me know if you prefer having the changes at the same PR or several dedicated ones ;)

Reply:
How would you feel about removing the complex filtering examples in this PR because, not that we really only have the one annotation, we really can only document the one filter and then doing a separate PR for `@Backwards` like you say?

Reply:
@lipsill I addressed the outdated section in #34050 already.